server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -171,199 +162,111 @@ static void clearLoadedModules()
+   * <p/>
+   * If user explicitly specifies druid.extension.toLoad, then it will look for those extensions under root
+   * extensions directory. If one of them is not found, druid will fail loudly.
+   * <p/>
+   * If user doesn't specify druid.extension.toLoad (or its value is empty), druid will load all the extensions
+   * under the root extensions directory.
+   *
+   * @param config ExtensionsConfig configured by druid.extensions.xxx
+   *
+   * @return an array of druid extension files that will be loaded by druid process
+   */
+  public static File[] getExtensionFilesToLoad(ExtensionsConfig config)
   {
-    URLClassLoader loader = loadersMap.get(coordinate);
-    if (loader == null) {
-      final CollectRequest collectRequest = new CollectRequest();
-
-      DefaultArtifact versionedArtifact;
-      try {
-        // this will throw an exception if no version is specified
-        versionedArtifact = new DefaultArtifact(coordinate);
-      }
-      catch (IllegalArgumentException e) {
-        // try appending the default version so we can specify artifacts without versions
-        if (defaultVersion != null) {
-          versionedArtifact = new DefaultArtifact(coordinate + ":" + defaultVersion);
-        } else {
-          throw e;
-        }
-      }
-
-      collectRequest.setRoot(new Dependency(versionedArtifact, JavaScopes.RUNTIME));
-      DependencyRequest dependencyRequest = new DependencyRequest(
-          collectRequest,
-          DependencyFilterUtils.andFilter(
-              DependencyFilterUtils.classpathFilter(JavaScopes.RUNTIME),
-              new DependencyFilter()
-              {
-                @Override
-                public boolean accept(DependencyNode node, List<DependencyNode> parents)
-                {
-                  if (accept(node.getArtifact())) {
-                    return false;
-                  }
-
-                  for (DependencyNode parent : parents) {
-                    if (accept(parent.getArtifact())) {
-                      return false;
-                    }
-                  }
-
-                  return true;
-                }
-
-                private boolean accept(final Artifact artifact)
-                {
-                  return exclusions.contains(artifact.getGroupId());
-                }
-              }
-          )
-      );
-
-      try {
-        final List<Artifact> artifacts = aether.resolveArtifacts(dependencyRequest);
-
-        List<URL> urls = Lists.newArrayListWithExpectedSize(artifacts.size());
-        for (Artifact artifact : artifacts) {
-          if (!exclusions.contains(artifact.getGroupId())) {
-            urls.add(artifact.getFile().toURI().toURL());
-          } else {
-            log.debug("Skipped Artifact[%s]", artifact);
-          }
-        }
-
-        for (URL url : urls) {
-          log.info("Added URL[%s]", url);
+    final File rootExtensionsDir = new File(config.getDirectory());
```

Comment:
The workflow doesn't explicitly check if it exists. if ```rootExtensionsDir``` doesn't exist, then ```listFiles()``` returns null, the return statement in this function handles that case.